app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -72,8 +72,8 @@ def distribution_repositories(host)
 
         if content_view && lifecycle_environment
           Katello::Repository.in_environment(lifecycle_environment).in_content_views([content_view]).
-              where(:distribution_version => host.os.release,
-                    :distribution_arch => host.architecture.name,
+              where(:distribution_version => host.try(:os).try(:release),
```

Comment:
I wonder if there might be any situation in which a host[group] somehow has an os but no arch?